app/components/ChartDateRangeModal.js

```diff
@@ -136,6 +147,25 @@ export const ChartDateRangeModal = (props) => {
     onClose();
   };
 
+  const handleDatesChange = newDates => {
+    const mostRecentDatumMoment = moment.utc(mostRecentDatumDate).tz(timezoneName);
+    const adjustedDates = setDateRangeToExtents(newDates);
+
+    // If the date selected contains the mostRecentDatum, we want to exclude the time
+    // between the mostRecentDatum and the end of that day. We also adjust the start
+    // time so that the period is a multiple of 24 hours.
+    if (mostRecentDatumMoment?.isBefore(adjustedDates?.endDate)) {
+      const hourCeiling = getLocalizedHourCeiling(mostRecentDatumDate, timePrefs);
+      const endDate = moment.utc(hourCeiling).tz(timezoneName);
+      const startDate = adjustedDates?.startDate?.hour(endDate.hour());
```

Comment:
In cases where a timezone change happens (say, a 1 hour shift) between the start and end dates, this would end up representing one more or one less hour than the expected multiple of 24 hours. 

If I'm thinking about this correctly (I may not be - correct me if I'm wrong), on a selection of 3 days in the spring here, if the change happened within the range, we would have 71 actual hours hrs represented.  Likewise, in the fall, we would have 73 hours.

Perhaps the startDate should be calculated by moment functions that take this into account?